### PR TITLE
Implement fastAPI prefix handling, run selenium tests against prefix

### PR DESCRIPTION
**Pull request by mvdbeek**
And fix a few prefix-related issues along the way.

EDIT: Skipped the ASGI static file serving for now, it caused the non_data_connection selenium test to fail in a bizarre way ...
~Also replaces the static WSGI middleware with a ASGI variant from starlette, modified to support our `_per_host` mechanism. I have not implemented the fine-grained per directory static mapping:~
```
    urlmap["/static"] = get_static_from_config("static_dir", "static/")
    urlmap["/images"] = get_static_from_config("static_images_dir", "static/images")
    urlmap["/static/scripts"] = get_static_from_config("static_scripts_dir", "static/scripts/")
    urlmap["/static/welcome.html"] = get_static_from_config("static_welcome_html", "static/welcome.html")
    urlmap["/favicon.ico"] = get_static_from_config("static_favicon_dir", "static/favicon.ico")
    urlmap["/robots.txt"] = get_static_from_config("static_robots_txt", "static/robots.txt")
```
~That seems like overkill when you can just exchange the static dir !?
There's a question mark about `/images` at the root path, we don't seem to use that anymore.
Wouldn't be hard to re-implement this finer-grained static mapping, let me know what you think about that.~

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy_test/base/populators.py [hunk 1]
```diff
@@ -1028,7 +1030,7 @@ def reimport_history(self, history_id, history_name, wait_on_history_length, exp
         download_path = self.export_url(history_id, export_kwds, api_key, check_download=True)
 
         # Create download for history
-        full_download_url = f"{url}{download_path}?key={api_key}"
+        full_download_url = urllib.parse.urljoin(download_path, f"?key={api_key}")
```
Comment (davelopez) [inline lib/galaxy_test/base/populators.py:1033]: Looks like the `url` parameter can now be dropped :)

**Review by davelopez — APPROVED** [review]
Looks good and works :)

File: client/src/libs/jquery/jquery.wymeditor.js [hunk 2]
```diff
@@ -28,24 +28,37 @@
    Namespace: WYMeditor
    Global WYMeditor namespace.
 */
-if(!WYMeditor) var WYMeditor = {};
+if (!WYMeditor) var WYMeditor = {};
```
Comment (dannon) [inline client/src/libs/jquery/jquery.wymeditor.js]: @mvdbeek Something went wrong with your prettier config here, this shouldn't have been touched?

File: .github/workflows/selenium.yaml [hunk 3]
```diff
@@ -7,6 +7,7 @@ on:
     paths-ignore:
       - 'doc/**'
 env:
+  GALAXY_CONFIG_GALAXY_URL_PREFIX: '/galaxypf'
```
Comment (jmchilton) [inline .github/workflows/selenium.yaml:10]: ❤️

**Review by jmchilton — APPROVED** [review]

File: client/src/libs/jquery/jquery.wymeditor.js [hunk 4]
```diff
@@ -317,380 +348,390 @@ jQuery.extend(WYMeditor, {
...
+                "</div>" +
+                "</div>",
+
+            logoHtml: "<a class='wym_wymeditor_link' " + "href='http://www.wymeditor.org/'>WYMeditor</a>",
+
+            iframeHtml:
+                "<div class='wym_iframe wym_section'>" +
+                "<iframe " +
+                "src='" +
+                "get_editor_iframe' " +
```
Comment (mvdbeek) [inline client/src/libs/jquery/jquery.wymeditor.js]: @dannon this was the change I had to make. I could just do this without prettier, or is that not looking like the right fix ?

**Review by dannon — APPROVED** [review]

**Pull request merged**